File: Trabajo2/CodigoEjercicio2.cs
using System;

public class Persona
{
    private string nombre, apellido, sexo;
    private uint edad;

    // Constructor
    public Persona()
    {
        nombre = "";
        apellido = "";
        sexo = "";
        edad = 0;
    }

    public void PedirDatos()
    {
        Console.WriteLine("\n=== INGRESO DE DATOS ===");
        
        Console.Write("Nombre: ");
        nombre = Console.ReadLine();

        Console.Write("Apellido: ");
        apellido = Console.ReadLine();

        Console.Write("Sexo (M/F): ");
        sexo = Console.ReadLine();

        Console.Write("Edad: ");
        string edadInput = Console.ReadLine();
        uint.TryParse(edadInput, out edad); // Conversión segura
    }

    public void MostrarDatos()
    {
        Console.WriteLine("\n=== DATOS DE LA PERSONA ===");
        Console.WriteLine("Nombre: " + nombre);
        Console.WriteLine("Apellido: " + apellido);
        Console.WriteLine("Sexo: " + sexo);
        Console.WriteLine("Edad: " + edad + " años");
        Console.WriteLine("============================");
    }
}

public class Program
{
    public static void Main(string[] args)
    {
        Persona[] personas = new Persona[2];

        Console.WriteLine("SISTEMA DE REGISTRO DE PERSONAS");
        Console.WriteLine("Ingrese los datos de 2 personas:");

        // Inicializar el array de personas
        for (int i = 0; i < personas.Length; i++)
        {
            personas[i] = new Persona();
        }

        // Pedir y mostrar datos
        for (int i = 0; i < personas.Length; i++)
        {
            Console.WriteLine("\nPersona " + (i + 1) + ":");
            personas[i].PedirDatos();
            personas[i].MostrarDatos();
        }

        Console.WriteLine("\nRegistro completado exitosamente!");
    }
}

File: Trabajo2/CodigoEjercicio 2.py
class Persona:
    def __init__(self):
        self.nombre = ""
        self.apellido = ""
        self.sexo = ""
        self.edad = 0

    def pedir_datos(self):
        print("\n=== INGRESO DE DATOS ===")
        
        self.nombre = input("Nombre: ")
        self.apellido = input("Apellido: ")
        self.sexo = input("Sexo (M/F): ")
        
        # Conversión segura de la edad
        edad_input = input("Edad: ")
        try:
            self.edad = int(edad_input)
            if self.edad < 0:
                self.edad = 0
        except ValueError:
            self.edad = 0

    def mostrar_datos(self):
        print("\n=== DATOS DE LA PERSONA ===")
        print(f"Nombre: {self.nombre}")
        print(f"Apellido: {self.apellido}")
        print(f"Sexo: {self.sexo}")
        print(f"Edad: {self.edad} años")
        print("============================")

def main():
    personas = [None] * 2

    print("SISTEMA DE REGISTRO DE PERSONAS")
    print("Ingrese los datos de 2 personas:")

    # Inicializar la lista de personas
    for i in range(len(personas)):
        personas[i] = Persona()

    # Pedir y mostrar datos
    for i in range(len(personas)):
        print(f"\nPersona {i + 1}:")
        personas[i].pedir_datos()
        personas[i].mostrar_datos()

    print("\nRegistro completado exitosamente!")

if __name__ == "__main__":
    main()